### PR TITLE
Add tests for Character semantics in regex matching

### PR DESCRIPTION
**Pull request by natecook1000**
Several of these tests are xfail'd, as we haven't implemented this
level of character semantics yet.

### CODE REVIEW

File: Tests/RegexTests/MatchTests.swift [hunk 1]
```diff
@@ -831,5 +831,156 @@ extension RegexTests {
     matchTest(#"(?<a>.)(.)\k<a>"#, input: "abac", match: "aba", xfail: true)
     matchTest(#"\g'+2'(.)(.)"#, input: "abac", match: "aba", xfail: true)
   }
+  
+  // MARK: Character Semantics
+  
+  var eComposed: String { "é" }
+  var eDecomposed: String { "e\u{301}" }
+
```
Comment (milseman) [inline Tests/RegexTests/MatchTests.swift:839]: - Try with both scalars being literals as well.
- Show group has a break requirement
- Show start/end of search may need boundary

File: Tests/RegexTests/MatchTests.swift [hunk 2]
```diff
@@ -831,5 +831,156 @@ extension RegexTests {
...
+    matchTest(#"[a-z][\u{300}-\u{320}]$"#, input: eDecomposed, match: eDecomposed,
+              xfail: true)
+
+    let flag = "🇰🇷"
+    matchTest(#"🇰🇷"#, input: flag, match: flag)
+    matchTest(#"[🇰🇷]"#, input: flag, match: flag)
+    matchTest(#"\u{1F1F0}\u{1F1F7}"#, input: flag, match: flag)
+    
+    // First Unicode scalar followed by CCC of regional indicators
+    matchTest(#"\u{1F1F0}[\u{1F1E6}-\u{1F1FF}]"#, input: flag, match: flag)
```
Comment (milseman) [inline Tests/RegexTests/MatchTests.swift:936]: - Test cases where you toss in a full Character or multi-scalar emoji into custom classes alongside scalar literals.

**Review by milseman — APPROVED** [review]
LGTM, some suggestions for other test cases to add

**Comment by natecook1000** [comment]
@swift-ci Please test

**Comment by natecook1000** [comment]
@swift-ci Please test Linux platform

**Comment by natecook1000** [comment]
@swift-ci Please test Linux platform

**Pull request merged**